[Fastlane] Added support for `for_platform` method, when parsing an Appfile

fabiomassimo opened the pull request:
# Summary

Improve integration with Fastlane by overriding `Appfile` configurations declared for a specific platform and/or lane.
# Why

An Appfile could include different configurations by specifying a platform and/or lane name.
Configurations are loaded accordingly to current driven lane in a specific platform (when specified).
# Example

Appfile as follow:

``` ruby
app_identifier "net.sunapps.1"
apple_id "felix@sunapps.net"
team_id "Q2CBPJ58CC"

for_lane :beta do
  app_identifier "net.sunapps.1.beta"
end

for_platform :ios do
  for_lane :enterprise do
    app_identifier 'com.app'
  end
end
```

The value inside the blocks will be used if matches with current context (platform, lane), otherwise the default configuration value will be loaded.

Commit:
Added platform support when parsing an "Appfile".

Commit:
Updated tests with new platform support when parsing an "Appfile".

KrauseFx commented:
This looks great, thanks for working on this @fabiomassimo. The only problem I can see right now is backwards compatibility: Right now users specified it like this:

```ruby
for_lane "ios beta" do
  app_identifier "net.sunapps.1.beta"
end
```

I think we'll need to support this notation as well and add a test for it to not break it in the future. Sorry about this :disappointed: 

Also, there seems to be some merge conflict.

Commit:
Merge branch 'master' of https://github.com/KrauseFx/CredentialsManager

Conflicts:
	lib/credentials_manager/appfile_config.rb
	spec/app_file_config_spec.rb

Commit:
Moved from sym reference, for blocks cached configurations, to string.
This will add support to "Appfile" that describes configuration for lane or platform specified as strings.

Commit:
Added test case.

KrauseFx commented:
Thank you for your contribution @fabiomassimo :rocket: :+1:
